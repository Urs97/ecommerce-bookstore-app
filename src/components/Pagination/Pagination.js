import React, { useContext } from 'react';
import './Pagination.scss';

import StoreContext from '../../context/StoreContext';

const Pagination = () => {
    const context = useContext(StoreContext);

    const currentPage = context.state.currentPage;
    const booksPerPage = context.state.booksPerPage;
    const totalBooks = context.currentData.length;
    const pageNumbers = [];

    console.log(context.currentData.length)

    for(let i = 1; i <= Math.ceil(totalBooks / booksPerPage); i++) {
        pageNumbers.push(i);
    };       

    return (
        <section className="pagination">     
            {pageNumbers.map(number => ( 
                <button 
                    key={number} 
                    disabled={currentPage === number ? true : false}
                    className={`${currentPage === number ? `button active-page` : `button`}`}
                    onClick={() => {context.changeCurrentPage(number)}}>{number}
                </button>
            ))}
        </section>
    );
};

export default Pagination;