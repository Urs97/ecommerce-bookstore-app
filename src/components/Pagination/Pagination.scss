@import '../../styles/shared.scss';

.pagination {
    display: flex;
    justify-content: center;
    margin-top: 1.5em;

    & .button {
        padding: 0.5em 0.75em;
        margin-right: 0.5em;
    }

    & .active-page {
        color: $green;
        background-color: #FFFFFF;
        box-shadow: inset 0 0 0 0.15em $green;
        cursor: auto;
    }
}