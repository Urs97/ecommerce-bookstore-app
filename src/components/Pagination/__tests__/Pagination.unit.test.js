import React from 'react';
import {shallow} from 'enzyme';
import Pagination from '../Pagination';

describe('Pagination', () => {
    test('renders empty Pagination', () => {
        const wrapper = shallow(
            <Pagination />
        );
        expect(wrapper).toMatchSnapshot();
    });
});