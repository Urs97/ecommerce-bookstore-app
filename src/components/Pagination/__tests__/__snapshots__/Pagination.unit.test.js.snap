// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders Pagination 1`] = `
<section
  className="pagination"
/>
`;
